### PR TITLE
Migrate Core to Python

### PR DESCRIPTION
**Pull request by jmcrumb**
Migrate core to python and implement several optimizations.  A few memory and code quality refactors, with the primary update being a performance optimization using python multithreading.

### CODE REVIEW

File: back-end/src/nova/main.go [hunk 1]
```diff
@@ -2,21 +2,35 @@ package main
...
+	nlp.Route(nlpGroup)
 
 	// route account database
+	accountGroup := router.Group("/account")
+	accounts.Route(accountGroup)
 
 	// route plugin store api
 
-	router.SetTrustedProxies([]string{"localhost"})
+	router.SetTrustedProxies([]string{"localhost", "127.0.0.1"})
```
Comment (joehartUSD) [inline back-end/src/nova/main.go:33]: I am unsure why 127.0.0.1 is used or would change beyond testing. Just curious though.

File: front-end/Nova/api/helper.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { AxiosProxyConfig } from 'axios';
+// import https from 'https';
+// import fs from 'fs';
+
+export const backEndSource: string = 'http://127.0.0.1:8080/';
+
+export const httpsProxy: AxiosProxyConfig = {
```
Comment (joehartUSD) [inline front-end/Nova/api/helper.ts:7]: What exactly is httpsProxy

File: nova-core/core/abstract_plugin.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    @abstractclassmethod
+    def execute_secondary_command(self, command: str) -> str:
+        pass
+
+    def request_history(self) -> list:
+        return [];
+    
+    def __string__(self) -> str: 
+        return __class__.__name__
```
Comment (joehartUSD) [inline nova-core/core/abstract_plugin.py:24]: "File Does Not End with a New Line"; is this an issue?

**Review by joehartUSD — APPROVED** [review]
Looks great to me. Seems like a solid platform to start user testing with. I just left some comments in the code for practice; feel free to answer them or not.

File: front-end/Nova/api/helper.ts (hunk 2 above)
Comment (jmcrumb) [reply front-end/Nova/api/helper.ts:7]: HTTPS proxy is a struct containing configuration settings for https

Comment (jmcrumb) [reply front-end/Nova/api/helper.ts:7]: HTTPS proxy is a struct containing configuration settings for https

File: nova-core/core/abstract_plugin.py (hunk 3 above)
Comment (jmcrumb) [reply nova-core/core/abstract_plugin.py:24]: for some types of files but not this type

Comment (jmcrumb) [reply nova-core/core/abstract_plugin.py:24]: for some types of files but not this type

**Pull request merged**